pkg/providers/provider.go

```diff
@@ -26,6 +26,7 @@ type Provider interface {
 	PostWorkloadInit(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec) error
 	BootstrapClusterOpts(clusterSpec *cluster.Spec) ([]bootstrapper.BootstrapClusterOption, error)
 	UpdateKubeConfig(content *[]byte, clusterName string) error
+	HardwareSpec() []byte
```

Comment:
How's `HardwareSpec()` get called? `GetRufioMachine` does't seem use it

Reply:
So, I've split the task into two PRs and this is the first, so it's not used here. The CLI workflows has access to an implementer of the `Provider` interface, and the `HardwareSpec` method will be called to retrieve the hardware spec to upload the file to git during the appropriate step.

Reply:
If this returns a `[]byte`, do we need to reevaluate how we use a method like `MachineConfigs` and generalize all to a `GetProviderSpec` for marshaling purposes and return `[]byte`? This way we can avoid introducing a new function every time we are dealing with a new type. But if it is too much work, I think the refactoring project @chrisdoherty4 is leading might take care of it.

Reply:
Yeah, I think that'd be too much effort for now. I did notice the "new workflows" that use hooks, which seems to be a good solution for this but it's currently feature gated.

Reply:
Discussed with @abhinavmpandey08. Will change `HardwareSpec()` -> `AdditionalFiles()`, so it makes more sense to be on the `Provider` type for now